frontend/src/metabase/modes/components/drill/ActionClickDrill/utils.ts

```diff
@@ -46,29 +50,25 @@ export function prepareParameter(
     return;
   }
 
-  return {
-    id: sourceParameterId,
-    type: actionParameter.type,
-    value: formatParameterValue(parameterValue),
-    target: actionParameterTarget,
-  };
+  return [actionParameter.id, formatParameterValue(parameterValue)];
 }
 
 function isMappedParameter(
   parameter: WritebackParameter,
-  parameterMappings: ParameterMappedForActionExecution[],
+  dashboardParamValues: ParametersForActionExecution,
 ) {
-  return parameterMappings.some(mapping =>
-    _.isEqual(mapping.target, parameter.target),
+  return Object.prototype.hasOwnProperty.call(
+    dashboardParamValues,
+    parameter.id,
   );
```

Comment:
this gets much more efficient because we can just check if the ID is a key in the map.